fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryLockFactory.java

```diff
@@ -58,43 +58,38 @@ private static class FDBDirectoryLock extends Lock {
 
         final AgilityContext agilityContext;
         final String lockName;
-        final long timeStampMillis;
+        long timeStampMillis;
         final int timeWindowMilliseconds;
         final byte[] fileLockKey;
         boolean closed;
+        private final Object fileLockSetLock = new Object();
         private static final Logger LOGGER = LoggerFactory.getLogger(FDBDirectoryLock.class);
 
         public FDBDirectoryLock(final AgilityContext agilityContext, final String lockName, byte[] fileLockKey, int timeWindowMilliseconds) {
             this.agilityContext = agilityContext;
             this.lockName = lockName; // for log messages
             this.fileLockKey = fileLockKey;
             this.timeWindowMilliseconds = timeWindowMilliseconds;
-            this.timeStampMillis = System.currentTimeMillis();
-            fileLockSet(timeStampMillis);
+            fileLockSet(false);
         }
 
         @Override
         public void close() {
-            fileLockClear(timeStampMillis);
+            fileLockClear();
             closed = true;
         }
 
         @Override
         public void ensureValid() {
```

Comment:
Looking at the call hierarchy, this is only used by `LockValidatingDirectoryWrapper`
Are we always using that, such that we can trust that any transaction doing a write would call `ensureValid`, or is it possible that it wouldn't validate, at the extreme, until it completes the merge.

Reply:
It looks like `IndexWriter` always wraps the directory, so we should be ok, but it could become a problem if more parts of the codec are changed to bypass the file-api and interact directly with `FDBDirectory` directly.

Reply:
I agree. There is a strong assumption that `ensureValid` is called periodically. If we'll identify a problem, a plan 'B' might be to hook an `ensureValid` call to the agilityContext.